designs/full-cluster-lifecycle-api-cloudstack.md

```diff
@@ -0,0 +1,81 @@
+# Cluster Lifecycle API CloudStack
+
+## Introduction
+
+### Problem
+
+Currently the only way to manage EKS Anywhere CloudStack cluster is by using CLI. We need to support native K8s experience on CloudStack cluster lifecycle.
+
+### Goals and Objectives
+
+* Create/Upgrade/Delete CloudStack workload clusters using: kubectl apply -f cluster.yaml
+* Create/Update/Upgrade/Delete CloudStack workload clusters using GitOps/Terraform
+
+#### Not in Scope
+
+* Create/Upgrade/Delete self-managed clusters with API
+
+#### Limitation
+
+* We currently only supports k8s version < 1.24 on CloudStack
+* We only support Redhat, not on Ubuntu or Bottlerocket yet.
+
+## Overview of Solution
+
+The EKS-A controller running in management cluster reconciles EKS-A CRDs, which fully manages workload clusters. [Detailed lifecycle diagram](https://github.com/aws/eks-anywhere/blob/4d307c6fe75075adadae38537d0f211c9142003e/designs/full-cluster-lifecycle-api.md#L55)
+
+### CloudStack defaults
+
+We already have [SetDefaults](https://github.com/aws/eks-anywhere/blob/ed4425dadb19600b4eb446d29b81f5c2441c16f6/pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go#L216) for datacenter, we can utilize it in its mutation webhook (need to create one). Cloudstack machine config doesn’t have machine config [default](https://github.com/aws/eks-anywhere/blob/ed4425dadb19600b4eb446d29b81f5c2441c16f6/pkg/providers/cloudstack/cloudstack.go#L162) values at this stage.
```

Comment:
Machines don't have a defaults method right now. But don't we have defaulting logic somewhere else? For example, in the provider "SetupAndBlahblah" methods

Reply:
No we don't have such defaulting logic